mmcls/models/backbones/tinyvit.py

```diff
@@ -0,0 +1,763 @@
+            else:
+                x = block(x)
+
+        if self.downsample is not None:
+            x = self.downsample(x)
+        return x
+
+
+@MODELS.register_module()
+class TinyViT(BaseBackbone):
+    """TinyViT.
+    A PyTorch implementation of : `TinyViT: Fast Pretraining Distillation
+    for Small Vision Transformers<https://arxiv.org/abs/2201.03545v1>`_
+
+    Inspiration from
+    https://github.com/microsoft/Cream/blob/main/TinyViT
+
+    Args:
+        arch (str | dict): The architecture of TinyViT.
+            Default: 'tinyvit_5m_224'.
+        in_channels (int): The number of input channels.
+            Default: 3.
+        depths (list[int]): The depth of each stage.
+            Default: [2, 2, 6, 2].
+        mlp_ratio (list[int]): The ratio of mlp hidden dim to embedding dim.
+            Default: 4.
+        drop_rate (float): Probability of an element to be zeroed.
+            Default: 0.
+        drop_path_rate (float): The drop path of the block.
+            Default: 0.1.
+        use_checkpoint (bool): Whether to use checkpointing to save memory.
+            Default: False.
+        mbconv_expand_ratio (int): The expand ratio of the mbconv.
+            Default: 4.0
+        local_conv_size (int): The size of the local conv.
+            Default: 3.
+        layer_lr_decay (float): The layer lr decay.
+            Default: 1.0
+        out_indices (int | list[int]): Output from which stages.
+            Default: -1
+        frozen_stages (int | list[int]): Stages to be frozen (all param fixed).
+            Default: -0
+        gap_before_final_nrom (bool): Whether to add a gap before the final
+            norm. Default: True.
+        act_cfg (dict): The activation config of the module.
+            Default: dict(type='GELU').
+        norm_cfg (dict): Config dict for normalization layer.
+            Default: dict(type='LN').
+        init_cfg (dict | list[dict], optional): Initialization config dict.
+            Default: None.
+    """
+    arch_settings = {
+        'tinyvit_5m_224': {
+            'resolution': (224, 224),
+            'channels': [64, 128, 160, 320],
+            'num_heads': [2, 4, 5, 10],
+            'window_sizes': [7, 7, 14, 7],
+        },
+        'tinyvit_11m_224': {
+            'resolution': (224, 224),
+            'channels': [64, 128, 256, 448],
+            'num_heads': [2, 4, 8, 14],
+            'window_sizes': [7, 7, 14, 7],
+        },
+        'tinyvit_21m_224': {
+            'resolution': (224, 224),
+            'channels': [96, 192, 384, 576],
+            'num_heads': [3, 6, 12, 18],
+            'window_sizes': [7, 7, 14, 7],
+        },
+        'tinyvit_21m_384': {
+            'resolution': (384, 384),
+            'channels': [96, 192, 384, 576],
+            'num_heads': [3, 6, 12, 18],
+            'window_sizes': [12, 12, 24, 12],
+        },
+        'tinyvit_21m_512': {
+            'resolution': (512, 512),
```

Comment:
Usually, the `resolution` (common name is `img_size`) and `window_sizes` should not be a part of the architecture, because they are closely related to the dataset samples.
And the `depths` should be a part of the architecture. Because it's the main architecture information, although it's the same among these archs.

Reply:
OK, I will update the settings with `depth`, while the window size is input size related in TinyViT, thus how to handle this issue? Keep the resolution in name(key) and remove the `resolution` ?